bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFieldImpl.java

```diff
@@ -124,16 +126,24 @@ public abstract class AbstractFieldImpl extends AbstractBaseImpl implements Fiel
     private Resource resource;
 
     @Override
-    @JsonInclude(JsonInclude.Include.NON_NULL)
     @Nullable
     public Boolean isReadOnly() {
+        return readOnly != null ? readOnly : Boolean.FALSE;
+    }
+
+    @JsonProperty("readOnly")
```

Comment:
This is added to keep the JSON payload minimal, we only add readOnly attribute if its present in JCR else we don't return anything